### DISCUSSION TITLE
Connect Heavy Node via VPN to an existing Master Node

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#8753] [Unsupported Versions] [answered]
Hello!

I plan to monitor the Internet uplink of a remote office with a seperate sensor and want to limit the traffic that gets send to my central Master Node over the VPN tunnel that is in place as the current WAN connection to this office. Is it a good idea to setup a Heavy Node at the remote location or does that have a huge impact on the overall performance when it comes to accessing the stored Elasticsearch data at the remote location? Any thoughts or actual experiences with this kind of setup?  Thanks in advance!

Cheers, Ben

**Comment by presianbg** [comment, marked as answer]
Hi @ben-sec,

We have 2-3 locations like this with Heavy Nodes connected to the HQ Manager via Wireguard VPN.
Everything works just fine! When the VPN goes down and you try to query the data, you will get shard errors and etc... so you have to filter the unreachable heavy nodes if happens to have such.

The thing is that the **only** the local data will be stored on the heavy node, so this is a big big + ! 

Cheers,
PY
